pallets/loans/src/lib.rs

```diff
@@ -833,37 +835,22 @@ impl<T: Config> Pallet<T> {
         T::PalletId::get().into_account()
     }
 
-    pub fn redeem_internal(
-        who: &T::AccountId,
-        currency_id: &CurrencyId,
-        redeem_amount: Balance,
-    ) -> DispatchResult {
-        let exchange_rate = Self::exchange_rate(currency_id);
-        let collateral = Self::calc_collateral_amount(redeem_amount, exchange_rate)?;
-        AccountDeposits::<T>::try_mutate_exists(currency_id, who, |deposits| -> DispatchResult {
-            let mut d = deposits.unwrap_or_default();
-            d.voucher_balance = d
-                .voucher_balance
-                .checked_sub(collateral)
-                .ok_or(ArithmeticError::Underflow)?;
-            if d.voucher_balance.is_zero() {
-                // remove deposits storage if zero balance
-                *deposits = None;
-            } else {
-                *deposits = Some(d);
-            }
-            Ok(())
-        })?;
-        TotalSupply::<T>::try_mutate(currency_id, |total_balance| -> DispatchResult {
-            let new_balance = total_balance
-                .checked_sub(collateral)
-                .ok_or(ArithmeticError::Underflow)?;
-            *total_balance = new_balance;
-            Ok(())
-        })?;
-        T::Currency::transfer(*currency_id, &Self::account_id(), who, redeem_amount)?;
-
-        Ok(())
+    fn get_account_liquidity(
+        account: &T::AccountId,
+    ) -> Result<(FixedU128, FixedU128), DispatchError> {
+        let total_borrow_value = Self::total_borrowed_value(account)?;
+        let total_collateral_value = Self::total_collateral_value(account)?;
+        if total_collateral_value > total_borrow_value {
+            Ok((
+                total_collateral_value - total_borrow_value,
```

Comment:
This line probably won't overflow in practice but maybe play safe and use `total_collateral_value.checked_sub(total_borrow_value).or_ok(ArithmeticError::Underflow)?` instead?

Reply:
I don't think it's necessary for It's an easy case.